1.cellprofiler_ic/image_quality_control/scripts/1.evaluate_qc.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Whole image quality control metric evaluation
+# 
+# In this notebook, we will use the outputted QC metrics to start working on developing thresholds using z-score to flag and skip images during CellProfiler illumination correction. We are evaluating plates 3-5 to find generalizable metrics and thresholds for this dataset.
+# 
+# **Blur metric to detect out of focus images** -> PowerLogLogSlope
+# 
+# **Saturation metric to detect large smudges** -> PercentMaximal
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+import numpy as np
+
+from scipy.stats import zscore
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+# ## Set paths and load in data frame
+
+# In[2]:
+
+
+# Directory for figures to be outputted
+figure_dir = pathlib.Path("./qc_figures")
+figure_dir.mkdir(exist_ok=True)
+
+# Directory with QC CellProfiler output per plate
+qc_dir = pathlib.Path("./qc_results")
+
+# Create an empty dictionary to store data frames for each plate
+all_qc_data_frames = {}
+
+# List all plates
+plates = ["Plate_3", "Plate_3_prime", "Plate_4", "Plate_5"]
+
+# Loop through each plate
+for plate in plates:
+    # Read in CSV with all image quality metrics per image for the current plate
+    qc_df = pd.read_csv(pathlib.Path(qc_dir / plate / "Image.csv"))
+
+    # Adding 'Metadata_Well' and 'Metadata_Site' columns using filename from FileName_OrigActin (same for all channels)
+    qc_df["Metadata_Well"] = qc_df["FileName_OrigRFP"].str.split("_").str[0]
```

Comment:
I am not sure what you mean here by expanding to multiple columns. I only need to use one of the channels FileName to generate the Well and Site (since not extracted during only QC pipeline) columns.